### PR TITLE
250 replace addpartforminputs with mui

### PR DESCRIPTION
**Pull request by Kodehodekatty**
### What is missing from the codebase?

Currently, we're using own styled input fields and selects in the add part components present in the add part page. To follow code structure we're using Mui Textfield component and Autocomplete component. 

### Changes made in this pull request:

- replaced inputs/selects with Mui's components. cleaned up in structure, fixed styling to fit overall design, removed old styling files or unused styles. 
- fixed type changes /removed old types/ removed "name" in template type
- updated zod validation based on types/ removed "template name" from validation
- removed template name input field
- added media query to finish button in add part page for phone screen.
- configured eslint file to ignore prettier end of line error that causes problems for Windows.

## Checklist

-   [x] **Review**: Have you reviewed your own changes to ensure they align with best practices?

### CODE REVIEW

File: .editorconfig
Comment (Benjamin-Lyon) [inline .editorconfig:1]: is this still needed if we are just going to ignore the endOfLine prettier rule instead?

File: src/components/AddPartFormFields/Type/Type.tsx [hunk 1]
```diff
@@ -31,32 +30,43 @@ export const Type: FC = () => {
...
-                <StyledIconContainer>
-                    <label htmlFor="type">Choose an item type</label>{' '}
-                    <ToolTip content="Specify Unit, or Item (lowest tier)">
-                        <FaRegQuestionCircleIcon />
-                    </ToolTip>
-                </StyledIconContainer>
-                <ErrorMessage
-                    name="type"
-                    render={({ message }) => <StyledErrorP>{message}</StyledErrorP>}
+            {' '}
```
Comment (Benjamin-Lyon) [inline src/components/AddPartFormFields/Type/Type.tsx]: We don't want these `{' '}` spaces

File: src/components/AddPartFormFields/styles.ts [hunk 2]
```diff
@@ -1,26 +1,27 @@
 import styled from 'styled-components';
 
 export const StyledInputWrap = styled.div`
-    display: flex;
-    flex-direction: row;
-    justify-content: space-between;
-    align-items: center;
-
-    flex-wrap: wrap;
+    display: grid;
```
Comment (Benjamin-Lyon) [inline src/components/AddPartFormFields/styles.ts]: can we use [flexbox](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Flexbox) to style here instead?

File: src/pages/addPart/styles.ts [hunk 3]
```diff
@@ -34,4 +35,10 @@ export const ButtonWrapper = styled.div`
     align-items: flex-end;
     justify-content: end;
     padding: 16px;
+    @media screen and (max-width: 800px) {
```
Comment (Benjamin-Lyon) [inline src/pages/addPart/styles.ts]: i like the use of media query here :)

**Review by Benjamin-Lyon — CHANGES_REQUESTED** [review]

**Review by Benjamin-Lyon — CHANGES_REQUESTED** [review]
see previous comments

**Review by Benjamin-Lyon — CHANGES_REQUESTED** [review]

File: src/components/AddItemFormFields/Type/Type.tsx
[the remark this answers is not in the document]
Comment (Kodehodekatty) [reply src/components/AddItemFormFields/Type/Type.tsx:1]: changed it to TemplateTypes

File: src/pages/addItem/Index.tsx [hunk 4]
```diff
@@ -74,7 +73,12 @@ const AddItem = () => {
             {location.pathname === `/add-item` ||
             location.pathname === '/add-item/add-form' ? null : (
                 <ButtonWrapper>
-                    <Button variant="black" onClick={handleNext}>
+                    <Button
```
Comment (Benjamin-Lyon) [inline src/pages/addItem/Index.tsx:76]: can we replace this button with the MUI button?

**Review by Benjamin-Lyon — CHANGES_REQUESTED** [review]

**Review by Benjamin-Lyon — APPROVED** [review]

**Pull request merged**